db/external_sst_file_ingestion_job.cc

```diff
@@ -390,34 +390,57 @@ Status ExternalSstFileIngestionJob::AssignLevelForIngestedFile(
     SuperVersion* sv, IngestedFileInfo* file_to_ingest, bool* overlap_with_db) {
   *overlap_with_db = false;
 
-  Arena arena;
-  ReadOptions ro;
-  ro.total_order_seek = true;
-
   Status status;
   int target_level = 0;
   auto* vstorage = cfd_->current()->storage_info();
+
+  // Always go to last level, fail if the file overlaps with existing files.
+  if (ingestion_options_.ingest_behind) {
+    // first check if new files overlaps with any file at the bottommost level
+    int bottom_lvl = cfd_->NumberLevels() - 1;
+    if (vstorage->NumLevelFiles(bottom_lvl) > 0) {
+      bool overlap_with_level = false;
+      status = IngestedFileOverlapWithLevel(sv, file_to_ingest, bottom_lvl,
```

Comment:
I think here you should call IngestedFileFitInLevel instead of IngestedFileOverlapWithLevel. Even if `overlap_with_level == false`, it is still possible that the file to be ingested cannot be put in that level because it may be fit in the hole of a file. For example, you have file1 with keys [1,2,3,9,10] on bottommost level. Now you are gonna ingest file2 with keys [3,4]. They don't overlap in terms of keys but it cannot be ingested because the key ranges overlap.

It could be better to add test cases to cover this case.